back/app/models/user.rb

```diff
@@ -558,15 +544,27 @@ def increment_confirmation_retry_count!
     save!
   end
 
+  private
+
+  def set_confirmation_required
+    return unless new_record? && email_changed?
```

Comment:
Good idea to do this here and keep all the logic together - but is the logic right? Should it be  `||` Don't we want to set confirmation to be required again every time email is changed or is that done elsewhere in the `new_email` logic?